tests/unit/test_transform_reporter.py

```diff
@@ -0,0 +1,180 @@
+        output = StringIO()
+
+        reporter = TransformReporter(
+            peak_allocations, format="gprof2dot", memory_records=[], native_traces=True
+        )
+
+        # WHEN
+        reporter.render_as_gprof2dot(output)
+        output.seek(0)
+
+        # THEN
+        output_data = json.loads(output.read())
+        assert output_data == {
+            "costs": [{"description": "Memory", "unit": "bytes"}],
+            "events": [{"callchain": [0], "cost": [1024]}],
+            "functions": [{"module": "fun.c", "name": "me"}],
+            "version": 0,
+        }
+
+    def test_multiple_allocations(self):
+        # GIVEN
+        peak_allocations = [
+            MockAllocationRecord(
+                tid=1,
+                address=0x1000000,
+                size=1024,
+                allocator=AllocatorType.MALLOC,
+                stack_id=1,
+                n_allocations=1,
+                _stack=[
+                    ("me", "foo.py", 12),
+                ],
+            ),
+            MockAllocationRecord(
+                tid=1,
+                address=0x1100000,
+                size=2048,
+                allocator=AllocatorType.VALLOC,
+                stack_id=2,
+                n_allocations=10,
+                _stack=[
+                    ("you", "bar.py", 21),
+                ],
+            ),
+        ]
+
+        output = StringIO()
+
+        reporter = TransformReporter(
+            peak_allocations, format="gprof2dot", memory_records=[], native_traces=False
+        )
+
+        # WHEN
+        reporter.render_as_gprof2dot(output)
+        output.seek(0)
+
+        # THEN
+        output_data = json.loads(output.read())
+        assert output_data == {
+            "costs": [{"description": "Memory", "unit": "bytes"}],
+            "events": [
+                {"callchain": [0], "cost": [1024]},
+                {"callchain": [1], "cost": [2048]},
+            ],
+            "functions": [
+                {"module": "foo.py", "name": "me"},
+                {"module": "bar.py", "name": "you"},
+            ],
+            "version": 0,
+        }
+
+    def test_empty_stack_trace(self):
+        # GIVEN
+        peak_allocations = [
+            MockAllocationRecord(
+                tid=1,
+                address=0x1000000,
+                size=1024,
+                allocator=AllocatorType.MALLOC,
+                stack_id=1,
+                n_allocations=1,
+                _stack=[],
+            ),
+        ]
+        output = StringIO()
+
+        reporter = TransformReporter(
+            peak_allocations, format="gprof2dot", memory_records=[], native_traces=False
+        )
+
+        # WHEN
+        reporter.render_as_gprof2dot(output)
+        output.seek(0)
+
+        # THEN
+        output_data = json.loads(output.read())
+        assert output_data == {
+            "costs": [{"description": "Memory", "unit": "bytes"}],
+            "events": [],
+            "functions": [],
```

Comment:
Hm - we don't want to report this case as `<unknown>` like we do elsewhere?

Reply:
Hummmm, that's an interesting point. Let me see what I can do.

Reply:
Hm. OK.